packages/pn-personafisica-webapp/src/pages/__test__/NotificationDetail.page.test.tsx

```diff
@@ -660,6 +660,68 @@ describe('NotificationDetail Page', async () => {
     vi.useRealTimers();
   });
 
+  it('should not duplicate payments when api call to cart respond with an error', async () => {
+    vi.useFakeTimers();
+    const paymentHistory = populatePaymentsPagoPaF24(
+      notificationToFe.timeline,
+      paymentsData.pagoPaF24,
+      paymentInfo
+    );
+    const requiredPaymentIndex = paymentHistory.findIndex(
+      (payment) => payment.pagoPa?.status === PaymentStatus.REQUIRED
+    );
+    const requiredPayment = paymentHistory[requiredPaymentIndex];
+    mock.onGet(`/bff/v1/notifications/received/${notificationDTO.iun}`).reply(200, notificationDTO);
+    mock.onPost(`/bff/v1/payments/info`, paymentInfoRequest.slice(0, 5)).reply(200, paymentInfo);
+    mock
+    .onPost(`/bff/v1/payments/cart`, {
+      paymentNotice: {
+        noticeNumber: requiredPayment.pagoPa?.noticeCode,
+        fiscalCode: requiredPayment.pagoPa?.creditorTaxId,
+        amount: requiredPayment.pagoPa?.amount,
+        companyName: notificationToFe.senderDenomination,
+        description: notificationToFe.subject,
+      },
+      returnUrl: window.location.href,
+    })
+    .reply(500);
+
+    expect(paymentHistory.length).toBe(6);
+
+    await act(async () => {
+      result = render(<NotificationDetail />, {
+        preloadedState: {
+          userState: { user: { fiscal_number: notificationDTO.recipients[2].taxId } },
+        },
+      });
+    });
+
+    const payButton = result.getByTestId('pay-button');
```

Comment:
here add the check that the payments stored in the redux state are 6.

TIPS: use testStore.getState().notificationState to get the notification detail state

Reply:
doone